src/ufo.rs

```diff
@@ -214,21 +214,20 @@ impl Ufo {
             // The v1 format stores some Postscript hinting related data in the lib,
             // which we only import into fontinfo if we're reading a v1 UFO.
             if meta.format_version == FormatVersion::V1 {
-                let mut feature_text = String::new();
                 let mut fontinfo =
                     if let Some(fontinfo) = font_info { fontinfo } else { FontInfo::default() };
 
+                let mut features_upgraded: Option<String> = None;
                 if let Some(lib_data) = &mut lib {
-                    upconversion::upconvert_ufov1_robofab_data(
+                    features_upgraded = upconversion::upconvert_ufov1_robofab_data(
                         &lib_path,
                         lib_data,
-                        &mut feature_text,
                         &mut fontinfo,
                     )?;
                 }
 
-                if !feature_text.is_empty() {
-                    features = Some(feature_text);
+                if features_upgraded.is_some() && !features_upgraded.as_ref().unwrap().is_empty() {
```

Comment:
I would write this as something like (and this may not compile, I haven't actually tried it):

```suggestion
                if features_upgraded.as_ref().map(String::is_empty).unwrap_or(false) {
```

but basically doing a `map` to get the bool, and then using `unwrap_or` to choose what happens in the `None` case.

Reply:
Hrm. I had to read that multiple times. It also compiles but makes the tests fail, so it has to be `!features_upgraded.as_ref().map(String::is_empty).unwrap_or(true)`. Uh... I'm going to go ahead and merge as is...